### PR TITLE
Windows plugins

### PR DESCRIPTION
**Pull request by farindk**
Implementation of plugin loading for Windows.

### CODE REVIEW

File: libheif/heif_init.cc [hunk 1]
```diff
@@ -22,11 +22,19 @@
 #include "heif.h"
 #include "error.h"
 #include "heif_plugin_registry.h"
+#include "common_utils.h"
 
 #if ENABLE_MULTITHREADING_SUPPORT
 #include <mutex>
 #endif
 
+#if defined(__MINGW32__) || defined(__MINGW64__) || defined(_MSC_VER)
```
Comment (kmilos) [inline libheif/heif_init.cc:31]: Just use `_WIN32` if this covers both MSVC and MinGW (i.e. is related to Windows as a platform and not a specific toolchain)... (This applies to all occurrences below btw.)

OTOH, this seems to be at odds with the `if(UNIX OR MINGW)` CMakeLists.txt above? Perhaps you want to invert the logic to `if(WIN32)`, or use something like `if(UNIX OR CYGWIN OR MSYS)` instead?

File: libheif/heif_init.cc [hunk 2]
```diff
@@ -22,11 +22,19 @@
...
+#include <windows.h>
```
Comment (kmilos) [inline libheif/heif_init.cc:32]: Seems to be included by plugins_windows.h already

File: libheif/heif_init.cc (hunk 1 above)
Comment (farindk) [reply libheif/heif_init.cc:31]: If I understand correctly, MINGW supports both, `dlopen` and `LoadLibrary`, right?
Then using `_WIN32` and `WIN32` in cmake would probably be the easiest variant, right?

Comment (kmilos) [reply libheif/heif_init.cc:31]: Nope, there is no dlopen on MinGW out of the box - there is implementations of it like dlfcn and similar wrappers ([like we discussed](https://github.com/strukturag/libheif/issues/731)), but you have to include the source or link another dep to your project (at which point probably MSVC can build it as well, which was the alternative way...)

So if you're doing the pure Win32 API like you decided, then it applies to both MSVC and MinGW equally...

(Cygwin and MSYS are different - they are POSIX **_emulation_** platforms, and ship all the POSIX calls as such... MinGW is "native" Windows)

Comment (farindk) [reply libheif/heif_init.cc:31]: Ok, thanks for the background info on this.
Let's try with simple WIN32 and _WIN32.

Comment (kmilos) [reply libheif/heif_init.cc:31]: Looking better, thanks! Still missing replacing w/ _WIN32 on this line though...

**Pull request merged**